client/src/components/CreateMealForm.js

```diff
@@ -0,0 +1,248 @@
+import React, { useState, useEffect, useContext } from "react";
+import { useForm } from "react-hook-form";
+import {
+  Typography,
+  TextField,
+  FormControl,
+  Button,
+  FormHelperText,
+  Box,
+} from "@material-ui/core";
+import { makeStyles } from "@material-ui/core/styles";
+import { DialogContext } from "contexts/DialogContext";
+import Dropzone from "common/DropZone";
+import API from "api";
+import imagePlaceholder from "images/imagePlaceholder.jpg";
+import { Context as AuthContext } from "contexts/AuthContext";
+
+const CreateMealForm = ({ meal }) => {
+  const mealId = meal ? meal.id : null;
+  const classes = useStyles();
+  let {
+    createMeal,
+    editMeal,
+    becomeChef,
+    state: { user },
+  } = useContext(AuthContext);
+
+  const [cuisines, setCuisines] = useState([]);
+  const [selectedCuisine, setSelectedCuisine] = useState(null);
+
+  useEffect(() => {
+    (async function getCuisines() {
+      const { data: allCuisines } = await API.get("/cuisines");
```

Comment:
Maybe we should be putting all the cuisines in a context so we don't need to keep re-fetching it everywhere